compiler/rustc_hir_typeck/messages.ftl

```diff
@@ -138,6 +139,11 @@ hir_typeck_option_result_asref = use `{$def_path}::as_ref` to convert `{$expecte
 hir_typeck_option_result_cloned = use `{$def_path}::cloned` to clone the value inside the `{$def_path}`
 hir_typeck_option_result_copied = use `{$def_path}::copied` to copy the value inside the `{$def_path}`
 
+hir_typeck_pass_to_variadic_function = can't pass `{$ty}` to variadic function
+    .suggestion = cast the value to `{$cast_ty}`
+    .help = cast the value to `{$cast_ty}`
+    .teach_help = certain types, like `{$ty}`, must be casted before passing them to a variadic function, because of arcane ABI rules dictated by the C standard
```

Comment:
It is not an ABI detail, it is a source language requirement: passing the argument triggers a _syntactic_ argument promotion.
```suggestion
    .teach_help = a `{$ty}` would be cast to `{$cast_ty}` by C's default argument promotion
```